CruCentralCoast/Database/CommunityGroup.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  CommunityGroup.swift
+//  CruCentralCoast
+//
+//  Created by Tyler Dahl on 7/25/18.
+//  Copyright © 2018 Landon Gerrits. All rights reserved.
+//
+
+import Foundation
+import FirebaseFirestore
+import RealmSwift
+
+class CommunityGroup: RealmObject {
+    
+    // Properties
+    @objc dynamic var id: String!
+    @objc dynamic var name: String!
+    @objc dynamic var summary: String?
+    @objc dynamic private var genderString: String!
+    @objc dynamic private var yearString: String!
+    @objc dynamic private var weekDayString: String!
+    @objc dynamic var time: String!
+    @objc dynamic var imageLink: String?
+    
+    // Relations
+    @objc dynamic var movement: Movement?
+    let leaders = List<Person>()
+    /// Inverse relationship that is auto-updated
+    let members = LinkingObjects(fromType: Person.self, property: "communityGroups")
+    
+    var gender: Gender { return Gender(rawValue: self.genderString) ?? .mixed }
+    var year: Year { return Year(rawValue: self.yearString) ?? .mixed }
+    var weekDay: WeekDay { return WeekDay(rawValue: self.weekDayString) ?? .sunday }
+    
+    func set(with dict: [String : Any]) {
+        guard let id = dict["id"] as? String,
```

Comment:
I'm not totally positive how the statement works, but will the whole thing return if only one of the let statements returns nil?

Reply:
Yes, although I should probably fatalError to help with debugging. Otherwise the object will still exist but have nil properties.